Userland/Libraries/LibURL/Parser.cpp

```diff
@@ -583,7 +583,8 @@ static bool ends_in_a_number_checker(StringView input)
         return true;
 
     // 5. If parsing last as an IPv4 number does not return failure, then return true.
-    if (parse_ipv4_number(last).has_value())
+    // NOTE: This is equivalent to checking that last is "0X" or "0x", followed by zero or more ASCII hex digits.
+    if ((last.starts_with("0X"sv) || last.starts_with("0x"sv)) && all_of(last.substring_view(2), is_ascii_hex_digit))
```

Comment:
You should be able to do `last.starts_with("0X"sv, CaseSensitivity::CaseInsensitive)` here.